packages/cli/src/modules/breaking-changes/breaking-changes.service.ts

```diff
@@ -0,0 +1,210 @@
+						if (workflowDetectionResult.isAffected) {
+							const affectedWorkflow = {
+								id: workflow.id,
+								name: workflow.name,
+								active: workflow.active,
+								issues: workflowDetectionResult.issues,
+							};
+							if (!allAffectedWorkflows.has(ruleId)) {
+								allAffectedWorkflows.set(ruleId, [affectedWorkflow]);
+							} else {
+								allAffectedWorkflows.get(ruleId)!.push(affectedWorkflow);
+							}
+						}
+					}
+				} catch (error) {
+					this.errorReporter.error(error);
+					this.logger.error('Rule detection failed', {
+						ruleId: rule.getMetadata().id,
+						error,
+					});
+				}
+			}
+		}
+
+		// Aggregate results
+		for (const rule of workflowLevelRules) {
+			const workflowResults = allAffectedWorkflows.get(rule.getMetadata().id) || [];
+			const isAffected = workflowResults.some((wr) => wr.issues.length > 0);
+
+			if (isAffected) {
+				allResults.push({
+					ruleId: rule.getMetadata().id,
+					affectedWorkflows: workflowResults,
+					instanceIssues: [],
+					recommendations: await rule.getRecommendations(workflowResults),
+				});
+			}
+		}
+		return allResults;
+	}
+
+	async refreshDetectionResults(
+		targetVersion: BreakingChangeVersion,
+	): Promise<DetectionReport | undefined> {
+		await this.cacheService.deleteFromHash(BreakingChangeService.CACHE_KEY_PREFIX, targetVersion);
+		return await this.getDetectionResults(targetVersion);
+	}
+
+	async getDetectionResults(
+		targetVersion: BreakingChangeVersion,
+	): Promise<DetectionReport | undefined> {
+		return await this.cacheService.getHashValue<DetectionReport>(
+			BreakingChangeService.CACHE_KEY_PREFIX,
+			targetVersion,
+			{
+				refreshFn: async () => {
+					return await this.detect(targetVersion);
+				},
+			},
+		);
+	}
+
+	async detect(targetVersion: BreakingChangeVersion): Promise<DetectionReport> {
+		const startTime = Date.now();
+		this.logger.info('Starting breaking change detection', { targetVersion });
+
+		const rules = this.ruleRegistry.getRules(targetVersion);
+
+		const workflowLevelRules = rules.filter((rule) => 'detectWorkflow' in rule);
+		const instanceLevelRules = rules.filter((rule) => 'detect' in rule);
+
+		// Get all workflow-level rules results
+		const workflowLevelResults = await this.getAllWorkflowRulesResults(workflowLevelRules);
+
+		// Run instance-level rules
+		const instanceLevelResults: DetectionResult[] = [];
+		for (const rule of instanceLevelRules) {
+			try {
+				const ruleResult = await rule.detect();
+				if (ruleResult.isAffected) {
+					instanceLevelResults.push({
+						ruleId: rule.getMetadata().id,
+						affectedWorkflows: [],
+						instanceIssues: ruleResult.instanceIssues,
+						recommendations: ruleResult.recommendations,
+					});
+				}
+			} catch (error) {
+				this.errorReporter.error(error);
+				this.logger.error('Rule detection failed', {
+					ruleId: rule.getMetadata().id,
+					error,
+				});
```

Comment:
IMO we shouldn't need to duplicate the logging when we use the error reporter. I guess that's not what happens now? We should probably change the error reporter a bit to enable both logging and reporting at the same time